### PR TITLE
Implement percentile function in keras.ops

### PR DESCRIPTION
**Pull request by shashaka**
## Description
Adds keras.ops.percentile, which computes the q-th percentile (0–100) of the input tensor along a specified axis.
Supported across NumPy, TensorFlow, PyTorch, and JAX backends. Not supported on OpenVINO.

## Contributor Agreement
**Please check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

> **Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

File: keras/src/ops/numpy.py [hunk 1]
```diff
@@ -6694,6 +6694,90 @@ def pad(x, pad_width, mode="constant", constant_values=None):
...
+        )
+
+    def compute_output_spec(self, x, q):
+        output_shape = reduce_shape(
+            x.shape, axis=self.axis, keepdims=self.keepdims
+        )
+
+        if hasattr(q, "shape"):
+            if len(q.shape) > 0:
+                output_shape = (q.shape[0],) + output_shape
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/ops/numpy.py:6718]: Jax only supports scalar or 1D q.

File: keras/src/ops/numpy.py [hunk 2]
```diff
@@ -6694,6 +6694,90 @@ def pad(x, pad_width, mode="constant", constant_values=None):
...
+        elif isinstance(q, (list, tuple)):
+            output_shape = (len(q),) + output_shape
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/ops/numpy.py:6720]: Jax only supports scalar or 1D q.

File: keras/src/backend/numpy/numpy.py [hunk 3]
```diff
@@ -1206,6 +1206,18 @@ def pad(x, pad_width, mode="constant", constant_values=None):
     return np.pad(x, pad_width, mode=mode, **kwargs)
 
 
+def percentile(x, q, axis=None, method="linear", keepdims=False):
+    x = convert_to_tensor(x)
+    q = convert_to_tensor(q)
+    ori_dtype = standardize_dtype(x.dtype)
+    if ori_dtype == "bool":
+        x = x.astype(config.floatx())
+    dtype = dtypes.result_type(x.dtype, float)
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/numpy/numpy.py:1221]: Updated.

File: keras/src/ops/numpy.py [hunk 4]
```diff
@@ -6694,6 +6694,90 @@ def pad(x, pad_width, mode="constant", constant_values=None):
...
+
+        if backend.is_int_dtype(x.dtype):
+            dtype = backend.floatx()
+        else:
+            dtype = dtypes.result_type(x.dtype, float)
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/ops/numpy.py]: Updated.

**Review by hertschuh — APPROVED** [review]
Thanks for adding this!
